packages/core/src/di/injector.ts

```diff
@@ -344,6 +344,8 @@ function resolveToken(
     }
   } else if (!(flags & InjectFlags.Self)) {
     value = parent.get(token, notFoundValue, InjectFlags.Default);
+  } else if (flags & InjectFlags.Self) {
```

Comment:
Redundant guard:

```suggestion
   } else {
```

Perhaps it would be better to swap this if with the preceding one.

Reply:
I like it! I think all this will soon be deprecated, so I made it into a separate condition with a specific problem.

Also it can be assigns a default value during initialization. What do you think about that?
```typescript
let value = notFoundValue;
```

Reply:
I prefer the explicitness of the additional `else` statement.